### DISCUSSION TITLE
Insane Layout shift even with CSS

### DISCUSSION THREAD

**Discussion by dukesx** [mantinedev/mantine#1925] [Q&A]
I am using v5.0.0 of mantine and using custom cache of emotion to server render Emotion in Next.JS. 

#### What i want 
I have a Grid, that i want to hide (display: none) on mobile phones but show on iPad and Above resolutions
#### The problem
When i open the website on mobile, due to late JS injection, the Cumulative Layout Shift (CLS) is FELT and SEEN , that is , i get a glimpse of Grid and then it hides as per CSS rules. This is NOT normal for Server Rendered CSS. I also have a tailwind CSS setup with Next JS to test if RAW CSS could fix it but it seems that i still see the glimpse of it and Light house seems to drive my CLS score to .9 (Which is pretty high)

#### Expected Behavior
I should not see the Grid at all provided that CSS is rendered on the server itself and my CLS score should be low. 

Any ideas what could be happening ? I am only experiencing this on V5.0 , before this version, I have never experienced this issue.

**Comment by rtivital** [comment]
You need to follow custom cache guide – https://mantine.dev/theming/emotion-cache/#server-side-rendering-with-custom-cache
I cannot provide any further assistance without reproduction.

**Reply by dukesx** [reply]
I am already using the custom cache setup but no luck. Grid comes and then hides. I have seen that others have also complained insane layout shifts with emotion. Not sure if related.

**Reply by rtivital** [reply]
Layout shift means that ssr does not work

**Reply by dukesx** [reply]
I was able to solve it. The problem was that i was using Tailwinds CSS file with Mantine. They don't operate at same speed. First Mantine injects styles then the style.css file (which you may use for external css overrides). This caused the Grid to show up for a brief second and then when the external css got resolved, the grid would disappear. Now i am using CreateStyles Api to do it and it works. 

This Explanation is for anyone who gets confused using a setup like Mine. External CSS (for overriding Mantine styles/ Tailwincss) + Mantine (Emotion JS).
